src/kemal/static_file_handler.cr

```diff
@@ -1,13 +1,94 @@
+require "zlib" ifdef !without_zlib
+
 module Kemal
-  # Kemal::StaticFileHandler is used to serve static files(.js/.css/.png e.g).
-  # This handler is on by default and you can disable it like.
-  #
-  #   serve_static false
-  #
+
   class StaticFileHandler < HTTP::StaticFileHandler
+    
     def call(context)
       return call_next(context) if context.request.path.not_nil! == "/"
-      super
+
+      unless context.request.method == "GET" || context.request.method == "HEAD"
```

Comment:
ok - this was actually taken from here

https://github.com/crystal-lang/crystal/blob/master/src/http/server/handlers/static_file_handler.cr#L21